node/node.go

```diff
@@ -435,18 +461,9 @@ func (n *Node) Stop() error {
 	// unblock n.Wait
 	close(n.stop)
 
-	// Remove the keystore if it was created ephemerally.
```

Comment:
@janos overall looks good to me, but I am not sure about the consequences of this modification to `Node.Stop()`. Looks like we might leave around this `ephermalKeystore` directory, as you are no longer removing it, and I don't know if this won't create issues.

Reply:
Note that `Node.Stop()` is used at many places throughout the codebase, and you've moved this logic to `Node.Close()`, but `Node.Close()` is not used at many places.

Reply:
Yes, you are correct completely. We can leave the ephemeral dir removal to Stop, since it is already broken (after stoping node with ephemeral storage it can not be reliably started), and this PR is not fixing that.

Reply:
This is now reverted, so this pr only adds close methods and call on them on network shutdown. I believe that it has much more chance to be accepted now, thank you.